spatial_maps/stats.py

```diff
@@ -179,3 +182,67 @@ def prob_dist_1d(x, bins):
 
     H, _ = np.histogram(x, bins=bins, normed=False)
     return (H / len(x)).T
+
+
+def _mask_in_both_if_any_is_nan(c0, c1):
+    """
+    Mask values in both cubes, if it is nan in cube 0 or/and
+    cube 1.
+    Returns boolean mask array.
+    """
+    bool_nan_c0 = np.isnan(c0)
+    bool_nan_c1 = np.isnan(c1)
+
+    mask_invalid = np.logical_or(bool_nan_c0,
+                                 bool_nan_c1)
+    mask_valid = ~mask_invalid
+    return mask_valid
+
+
+def population_vector_correlation(rmaps1, rmaps2, min_rate=None):
+    """
+    Calcualte population vector correlation between two
+    stacks of rate maps.
+    
+    Parameters
+    ----------
+    rmaps1 : ndarray
+    Array of the shape [n_units, n_bins_dim1, n_bins_dim2]
+    rmaps2 : ndarray
+    Array of the shape [n_units, n_bins_dim1, n_bins_dim2]
+
+    Returns
+    -------
+    pop_vec_corr : float
+    Population vector correlation
+    """
+    # make is float
+    rmaps1 = rmaps1.astype(float)
+    rmaps2 = rmaps2.astype(float)
+    
+    bins_x = rmaps1.shape[1]
+    bins_y = rmaps1.shape[2]
+
+    assert rmaps1.shape == rmaps2.shape
+    # correlation coefficient requires at least 2x2 values
+    assert rmaps1.shape[0] > 1
+
+    corr_coeff = np.zeros((bins_x, bins_y))
+    corr_coeff[:] = np.nan
+    
+    # create mask. exclude value if nan in any of the two stacks
+    mask_valid = _mask_in_both_if_any_is_nan(rmaps1, rmaps2)
```

Comment:
I think this should be off by default - to notify the user that there are nan in the stack

Reply:
maybe put a mask_nan=False as an argument?

Reply:
good point. made it also slightly more concise, dropped the extra function for the nan detection